src/functions.rs

```diff
@@ -178,33 +197,27 @@ pub fn is_executable(path: impl AsRef<Path>) -> io::Result<bool> {
 pub fn access_syscall(path: impl AsRef<Path>, mode_mask: c_int) -> io::Result<bool> {
     let path = path.as_ref();
 
-    // Syscall for unix and windows systems: https://stackoverflow.com/a/59224987/9982477
-    let mut buf = Vec::new();
-    let buf_ptr;
     #[cfg(unix)]
-    {
-        use std::os::unix::ffi::OsStrExt;
-        buf.extend(path.as_os_str().as_bytes());
-        buf.push(0u8);
-        buf_ptr = buf.as_ptr() as *const libc::c_char;
+    fn access(path: &Path, mode_mask: c_int) -> i32 {
+        use std::os::unix::prelude::OsStrExt;
+
+        let cstr = CString::new(path.as_os_str().as_bytes()).expect("Path had interior nul byte");
```

Comment:
Another alternative is to "silently" fail if there's an interior nul byte (which is the current behavior anyway)

```rust
    #[cfg(unix)]
    fn access(path: &Path, mode_mask: c_int) -> i32 {
        use std::os::unix::prelude::OsStrExt;

        let bytes = path.as_os_str().as_bytes().to_owned();
        let cstr = unsafe { CString::from_vec_unchecked(bytes) };
        unsafe { libc::access(cstr.as_ptr(), mode_mask) }
    }
```

Reply:
Both are cool, so I'll resolve this and leave as is.